arch/Makefile.rules

```diff
@@ -19,3 +19,10 @@ endef
 define aarch64_no_reserved_tcb_overlap =
 	$(eval LIBCONTEXT_CFLAGS-y+=-DCONFIG_AARCH64_NO_TCB_OVERLAP=1)
 endef
+
+# Converts an architecture string to the corresponding name of a configuration variable
+# Usage:
+#  $(call ukarch_str2cfg,"<architecture>")
+define ukarch_str2cfg =
+$(addprefix CONFIG_ARCH_,$(call uc,$(subst arm,ARM_32,$(subst arm64,ARM_64,$(1)))))
```

Comment:
This concerns me a bit wrt future architectures eg RISC, as it's essentially a bug lurking. I know it won't be as terse, but it would be more fail-safe if this used a controlled vocabulary for all known architectures and issue an error if an unknown arch passed. Perhaps it could be implemented as a separate macro that checks against known architectures.

Reply:
At the moment, external architectures are not supported, in this sense this won't break the status quo. Clearly it might need to get modified if we know how we support external architectures. However, the function is only handling the special cases of Arm where we have an extra `_` in the config option: `CONFIG_ARCH_ARM_32` although we call the architecture `arm32`. My intention was that this function is future-proof if we match the config name with the architecture name (like we do on x86): For example, if we name Risc-V `riscv`, then the matching architecture configuration option must be `CONFIG_ARCH_RISCV`.
What do you mean with separate macros?

Reply:
> At the moment, external architectures are not supported, in this sense this won't break the status quo. Clearly it might need to get modified if we know how we support external architectures

Yup, I was not referring to external architectures, as we already agreed that may not even be possible, but to new internal architectures.

> My intention was that this function is future-proof if we match the config name with the architecture name (like we do on x86

That expectation is exactly what I am concerned about. problem If the author of a new architecture is not be aware of this expectation (and how would they be?) then this feature will be broken on the new architecture. An explicit check, on the other hand, will reveal such an issue right away.

Reply:
Hum... my concern is that a different expression which is extensible would be too complex for now and we also do not know yet how we will include external archs. Maybe there will be a more natural solution to the problem.
Example:
```Makefile
ukarch_str2cfg = $(addprefix CONFIG_ARCH_,$(CONFIG_ARCH_MAP_$(1)))
```
...which requires that each arch sets a special var:
```Makefile
CONFIG_ARCH_MAP_x86_64=X86_64
CONFIG_ARCH_MAP_arm32=ARM_32
CONFIG_ARCH_MAP_arm64=ARM_64
```
Additionally, code needs to be added to throw an error when the map variable is not defined.

However, I can comment what the function assumes so that this information helps as soon as we touch the code again.

Reply:
Works for me, thanks 👍🏼

Reply:
👍🏻